### PR TITLE
Add support for `make verify WHAT=typecheck`

### PR DESCRIPTION
**Pull request by rmmh**
This will be used to make the typecheck job emit junit and avoids duplicating the rest of the verify logic.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: hack/make-rules/verify.sh [hunk 1]
```diff
@@ -109,13 +120,19 @@ function run-checks {
...
   do
     local check_name="$(basename "${t}")"
-    if is-excluded "${t}" ; then
-      echo "Skipping ${check_name}"
-      continue
-    fi
-    if ${QUICK} && ! is-quick "${t}" ; then
-      echo "Skipping ${check_name} in quick mode"
-      continue
+    if [[ ! -z ${VERIFY_CHECKS} ]]; then
```
Comment (BenTheElder) [inline hack/make-rules/verify.sh]: make this `${VERIFY_CHECKS:-}` since we set nounset

Comment (rmmh) [reply hack/make-rules/verify.sh]: Done.

File: hack/make-rules/verify.sh [hunk 2]
```diff
@@ -73,6 +73,17 @@ function is-quick {
   return 1
 }
 
+function is-explicitly-chosen {
+  local name="${1#verify-}"
+  name="${name%.*}"
+  for e in ${VERIFY_CHECKS}; do
+    if [[ $e == $name ]]; then
```
Comment (ixdy) [inline hack/make-rules/verify.sh]: nit: we generally use `${var}` form and quote variables.

File: hack/make-rules/verify.sh [hunk 3]
```diff
@@ -73,6 +73,17 @@ function is-quick {
   return 1
 }
 
+function is-explicitly-chosen {
+  local name="${1#verify-}"
+  name="${name%.*}"
+  for e in ${VERIFY_CHECKS}; do
```
Comment (ixdy) [inline hack/make-rules/verify.sh]: maybe `VERIFY_CHECKS` should be documented somewhere, e.g. in the `make verify` help text?

File: hack/make-rules/verify.sh (hunk 2 above)
Comment (rmmh) [reply hack/make-rules/verify.sh]: you don't need to quote variables in [[ blocks

Comment (cblecker) [reply hack/make-rules/verify.sh]: https://github.com/koalaman/shellcheck/wiki/SC2053

I agree with @ixdy that we should use the `"${name}"` here

File: hack/make-rules/verify.sh (hunk 3 above)
Comment (cblecker) [reply hack/make-rules/verify.sh]: yeah, I think it should be [documented](https://github.com/kubernetes/kubernetes/blob/209cdd90489a5af11f2f124e592c57df82a3da28/build/root/Makefile#L110-L127). I'm also wondering if we should follow the `WHAT=` pattern that a lot of other makerules follow, and have WHAT map in this case to the VERIFY_CHECKS var, like BRANCH already does to KUBE_VERIFY_GIT_BRANCH

Comment (BenTheElder) [reply hack/make-rules/verify.sh]: WHAT= SGTM

Comment (rmmh) [reply hack/make-rules/verify.sh]: Done.

File: hack/make-rules/verify.sh (hunk 2 above)
Comment (rmmh) [reply hack/make-rules/verify.sh]: Done.

**Comment by cblecker** [comment]
/lgtm
/approve
/sig testing

@jdumars This PR isn't a bugfix, but it will enable us to turn up a new (currently non-blocking) typecheck presubmit that we are hoping prevents future cross build failures. Any objections to this being merged?

@spiffxp @fejta @timothysc @stevekuznetsov If Jaice has no objections, could you `/status approved-for-milestone` for sig-testing?

**Comment by jdumars** [comment]
LGTM! Ship it!

**Comment by krzyzacy** [comment]
/status approved-for-milestone
per LGTM from @jdumars

**Comment by BenTheElder** [comment]
I'll update the job

**Pull request merged**